evm-contracts/src/v0.7/dev/CompoundPriceFlaggingValidator.sol

```diff
@@ -0,0 +1,277 @@
+
+  event CompoundOpenOracleAddressUpdated(
+    address indexed from,
+    address indexed to
+  );
+  event FlagsAddressUpdated(
+    address indexed from,
+    address indexed to
+  );
+  event ThresholdUpdated(
+    address indexed aggregator,
+    string symbol,
+    uint8 decimals,
+    uint32 deviationThresholdDenominator
+  );
+  
+  /**
+   * @notice Create a new CompoundPriceFlaggingValidator
+   * @dev Use this contract to compare Chainlink aggregator prices
+   * against the Compound Open Oracle prices
+   * @param flagsAddress Address of the flag contract
+   * @param compoundOracleAddress Address of the Compound Open Oracle UniswapAnchoredView contract
+   */
+  constructor(address flagsAddress, address compoundOracleAddress)
+    ConfirmedOwner(msg.sender)
+  {
+    setFlagsAddress(flagsAddress);
+    setCompoundOpenOracleAddress(compoundOracleAddress);
+  }
+
+  /**
+   * @notice Set the address of the Compound Open Oracle UniswapAnchoredView contract
+   * @param oracleAddress Compound Open Oracle UniswapAnchoredView address
+   */
+  function setCompoundOpenOracleAddress(address oracleAddress)
+    public
+    onlyOwner()
+  {
+    address previous = address(s_compOpenOracle);
+    if (previous != oracleAddress) {
+      s_compOpenOracle = UniswapAnchoredView(oracleAddress);
+      emit CompoundOpenOracleAddressUpdated(previous, oracleAddress);
+    }
+  }
+
+  /**
+   * @notice Updates the flagging contract address for raising flags
+   * @param flagsAddress sets the address of the flags contract
+   */
+  function setFlagsAddress(address flagsAddress)
+    public
+    onlyOwner()
+  {
+    address previous = address(s_flags);
+    if (previous != flagsAddress) {
+      s_flags = FlagsInterface(flagsAddress);
+      emit FlagsAddressUpdated(previous, flagsAddress);
+    }
+  }
+
+  /**
+   * @notice Set the threshold details for comparing a Chainlink aggregator
+   * to a Compound Open Oracle feed.
+   * @param aggregator The Chainlink aggregator address
+   * @param compoundSymbol The symbol used by Compound for this feed
+   * @param compoundDecimals The number of decimals in the Compound feed
+   * @param compoundDeviationThresholdDenominator The threshold denominator use to determine
+   * the percentage with which the difference in prices must reside within. For example:
+   * If the prices are valid within a 5% threshold, and 1 / denominator, this denominator
+   * = 20, since 1/20 = 5.
+   */
+  function setThreshold(
+    address aggregator,
+    string calldata compoundSymbol,
+    uint8 compoundDecimals,
+    uint32 compoundDeviationThresholdDenominator
+  ) 
+    public 
+    onlyOwner() 
+  {
+    CompoundFeedDetails memory compDetails = s_thresholds[aggregator];
+    compDetails.symbol = compoundSymbol;
+    compDetails.decimals = compoundDecimals;
+    compDetails.deviationThresholdDenominator = compoundDeviationThresholdDenominator;
+    s_thresholds[aggregator] = compDetails;
```

Comment:
```suggestion
    s_thresholds[aggregator] = CompoundFeedDetails({
      symbol: compoundSymbol,
      decimals: compoundDecimals,
      deviationThresholdDenominator: compoundDeviationThresholdDenominator
    });
```
Looks like you're overwriting every field on the struct. No need to read if from storage only to clobber it, just clobber it right off the bat and save some gas.